Added ItemGraph/remap_data guide and advanced ItemGraph sample notebook

nparavicini7 opened the pull request:
Added a beginner/intermediate guide on the ItemGraph, remap_data(), and how to use them together, and then added a sample notebook that users can download to maintain their own organization-wide dependency graph and use it for analysis/admin workflows. Thank you @ManushiM for the section on automating it

ManushiM commented on samples/03_org_administrators/org_wide_dependency_graph.ipynb:
```diff
@@ -0,0 +1,1264 @@
+{
```
<p>Replace <em>Sorting of org items </em>with <em>Sorting of items in our organization</em></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/2311/discussion/#comment-2193902648'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

ManushiM replied:
<p>Might help to add a line or so about <code>.gml</code> files for the beginner user, or link to some resource to learn more.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/2311/discussion/#comment-2193902650'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

ManushiM replied:
<p>Same thing for <code>networkx</code>. Some users may not know it's a third party library, and a line and link may help.   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/2311/discussion/#comment-2193902653'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

nparavicini7 commented:
<p>Addressed 2 and 3. For 1, no, should we provide a list of the items compatible with remap_data()? There's only a subset currently. We could do an asterisk and list at the bottom or something </p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/2311/discussion/#comment-3049940220'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nparavicini7 commented:
Addressed all of your comments @jyaistMap @ManushiM. The one thing I didn't add was a list of the compatible JSON-based items for remap_data()- I think that would be more appropriate for the docstring of the function

jyaistMap commented:
> Addressed all of your comments @jyaistMap @ManushiM. The one thing I didn't add was a list of the compatible JSON-based items for remap_data()- I think that would be more appropriate for the docstring of the function

Good idea. Sounds appropriate.
